pkg/catalogv2/git/git.go

```diff
@@ -1,480 +1,395 @@
+
+	// List Options
+	r.listOpts.InsecureSkipTLS = r.insecureTLSVerify
+
+	// Reset Options
+	r.resetOpts.Mode = gogit.HardReset
+
+	// Set authentication Methods for cloning
+	if r.auth != nil {
+		r.cloneOpts.Auth = r.auth
+		r.fetchOpts.Auth = r.auth
+		r.listOpts.Auth = r.auth
+	}
+	// Set CABundle for all git operations
+	if len(r.caBundle) > 0 {
+		r.cloneOpts.CABundle = r.caBundle
+		r.fetchOpts.CABundle = r.caBundle
+		r.listOpts.CABundle = r.caBundle
+	}
+	// Debug if enabled
+	if logrus.IsLevelEnabled(logrus.DebugLevel) {
+		r.cloneOpts.Progress = os.Stdout
+		r.fetchOpts.Progress = os.Stdout
 	}
+}
 
-	client, err := g.httpClientWithCreds()
-	if err != nil {
-		logrus.Warnf("Problem creating http client to check git remote sha of repo [%v]: %v", g.URL, err)
-		return true, nil
+// cloneOrOpen executes the clone operation of a git repository at given branch with depth = 1 if it does not exist.
+// If exists, just open the local repository and assign it to ro.localRepo
+func (r *Repository) cloneOrOpen(branch string) error {
+	var err error
+	cloneOptions := r.cloneOpts
+	if branch != "" {
+		cloneOptions.ReferenceName = plumbing.NewBranchReferenceName(branch)
 	}
-	defer client.CloseIdleConnections()
 
-	req, err := http.NewRequest("GET", formattedURL, nil)
+	err = cloneOptions.Validate()
 	if err != nil {
-		logrus.Warnf("Problem creating request to check git remote sha of repo [%v]: %v", g.URL, err)
-		return true, nil
+		return fmt.Errorf("plainClone validation failure: %w", err)
 	}
 
-	req.Header.Set("Accept", "application/vnd.github.v3.sha")
-	req.Header.Set("If-None-Match", fmt.Sprintf("\"%s\"", sha))
-	for k, v := range g.headers {
-		req.Header.Set(k, v)
-	}
-
-	resp, err := client.Do(req)
-	if err != nil {
-		// Return timeout errors so caller can decide whether or not to proceed with updating the repo
-		uErr := &url.Error{}
-		if ok := errors.As(err, &uErr); ok && uErr.Timeout() {
-			return false, errors.Wrapf(uErr, "Repo [%v] is not accessible", g.URL)
+	err = r.plainOpen()
+	if err == gogit.ErrRepositoryNotExists {
+		r.repoGogit, err = gogit.PlainClone(r.Directory, false, cloneOptions)
+		if err != nil && err != gogit.ErrRepositoryAlreadyExists {
+			return fmt.Errorf("plainClone failure: %w", err)
 		}
-		return true, nil
+		return nil
 	}
-	defer resp.Body.Close()
-
-	if resp.StatusCode == http.StatusNotModified {
-		return false, nil
+	if err != nil {
+		return fmt.Errorf("plainOpen failure: %w", err)
 	}
 
-	return true, nil
+	return nil
 }
 
-func (g *git) git(args ...string) error {
-	var output io.Writer
-	if logrus.IsLevelEnabled(logrus.DebugLevel) {
-		output = os.Stdout
+// plainOpen opens an existing local git repository on the specified folder not walking parent directories looking for '.git/'
+func (r *Repository) plainOpen() error {
+	openOptions := gogit.PlainOpenOptions{
+		DetectDotGit: false,
+	}
+	localRepository, err := gogit.PlainOpenWithOptions(r.Directory, &openOptions)
```

Comment:
Same recommendation as for `PlainClone`, and use `Open` instead.

Reply:
Same answer above as for in PlainClone, it will be changed in another issue